src/ConnectedMode/BranchMatcher.cs

```diff
@@ -51,64 +52,85 @@ public interface IBranchMatcher
     internal class BranchMatcher : IBranchMatcher
     {
         private readonly ISonarQubeService sonarQubeService;
+        private readonly ILogger logger;
 
         [ImportingConstructor]
-        public BranchMatcher(ISonarQubeService sonarQubeService)
+        public BranchMatcher(ISonarQubeService sonarQubeService, ILogger logger)
         {
             this.sonarQubeService = sonarQubeService;
+            this.logger = logger;
         }
 
         public async Task<string> GetMatchingBranch(string projectKey, IRepository gitRepo, CancellationToken token)
         {
             Debug.Assert(sonarQubeService.IsConnected,
                 "Not expecting GetMatchedBranch to be called unless we are in Connected Mode");
+
+            logger.LogVerbose(Resources.BranchMapper_CalculatingServerBranch_Started);
```

Comment:
@ugras-ergun-sonarsource apologies, I forgot to add review comments to this PR.

I moved all of the code that was inside the try-catch to a separate method.
There was no change to existing logic except to add the logging.

The main reason was to make it easier to add a single final log message showing which branch was chosen.